### PR TITLE
feat: top tabbar & bottom tabbar

### PR DESCRIPTION
**Pull request by friendlysxw**
## Summary by CodeRabbit

- **New Features**
  - Introduced new `PageTabBar` for enhanced navigation.
  - Added `PageContent` component for consistent page layout styling.
  - Enhanced header components with `HeaderCenter`.
  
- **Refactor**
  - Simplified tab layout and rendering logic across various components.
  - Updated activity and explore components to use new layout and tab bar structures.

- **Style**
  - Adjusted default tab bar height for a more compact design.
  - Introduced new boxShadow variant `"md-top"` for specific use cases.

- **Bug Fixes**
  - Corrected layout calculation in `ExploreStyled` for better alignment.

### CODE REVIEW

File: app/(tabs)/_layout.tsx [hunk 1]
```diff
@@ -76,27 +63,10 @@ export default function TabLayout() {
...
-                  <UserGlobalPoints />
-                  <SearchLink />
-                  {/* <PostLink />
-                  <View>
-                    <ExploreSharingButton fid={currFid} />
-                  </View> */}
-                </HeaderRight>
-              </Header>
-            ),
+            headerShown: false,
```
Comment (sin-bufan) [inline app/(tabs)/_layout.tsx:69]: 这个headerShown应该可以统一加在Tabs组件上，不用分开每个都加

**Review by sin-bufan — APPROVED** [review]

File: app/(tabs)/_layout.tsx (hunk 1 above)
Comment (friendlysxw) [reply app/(tabs)/_layout.tsx:69]: 这个我目前只改了我做的页面，等页面都改完了再统一在Tabs上加吧

File: components/layout/tabBar/TabBar.tsx [hunk 2]
```diff
@@ -60,13 +65,13 @@ function MobileTabBar(props: BottomTabBarProps) {
               {options.tabBarIcon &&
                 options.tabBarIcon({
                   focused: isFocused,
-                  color: isFocused ? "#A36EFE" : "#4C2896",
+                  color: isFocused ? "#FFF" : "#A36EFE",
```
Comment (sin-bufan) [inline components/layout/tabBar/TabBar.tsx:68]: 这里的颜色，要不要在constants里面加2个常量来调用，下一版要改ui的话也不用逐个去改主题色了
export const PRIMERY_COLOR = "#4C2896";
export const SECONDARY_COLOR = "#A36EFE";

**Review by sin-bufan — APPROVED** [review]

File: components/layout/tabBar/TabBar.tsx (hunk 2 above)
Comment (friendlysxw) [reply components/layout/tabBar/TabBar.tsx:68]: OK，我改下

**Pull request merged**